### ISSUE TITLE
Tooltips show erroneous values when used with mark level sort

### ISSUE THREAD

**Issue by Saba9** [vega/vega#1390]
Values displayed by the tooltip don't correspond to point being hovered over:
```json
{
  "$schema": "https://vega.github.io/schema/vega/v4.json",
  "width": 300,
  "height": 200,
  "style": "cell",
  "data": [
    {
      "name": "source_0",
      "values": [
        {"group": 0, "count": 100},
        {"group": 1, "count": 150},
        {"group": 2, "count": 250}
      ]
    }
  ],
  "marks": [
    {
      "name": "marks",
      "type": "area",
      "sort": {"field": "datum.group", "order": "descending"},
      "from": {"data": "source_0"},
      "encode": {
        "update": {
          "orient": {"value": "vertical"},
          "tooltip": {
            "signal": "{group: datum.group, count: datum.count}"
          },
          "x": {"scale": "x", "field": "group"},
          "y": {"scale": "y", "field": "count"},
          "y2": {"scale": "y", "value": 0}
        }
      }
    }
  ],
  "scales": [
    {
      "name": "x",
      "type": "linear",
      "domain": {"data": "source_0", "field": "group"},
      "range": [0, {"signal": "width"}]
    },
    {
      "name": "y",
      "type": "linear",
      "domain": {"data": "source_0", "field": "count"},
      "range": [{"signal": "height"}, 0]
    }
  ]
}
```
Note that changing the sort to ascending order fixes the tooltips. Noticed this issue while looking at [this vega-lite example](https://vega.github.io/editor/#/examples/vega-lite/area).

**Comment by jheer** [comment]
Thanks! This seems to be a bug specific to area marks, where our [binary search procedure to find the tooltip point](https://github.com/vega/vega-scenegraph/blob/master/src/util/pickPath.js#L1) assumes an ascending order. A simple check + reversal should suffice to handle descending sort, but we might want to ensure correct picking even for "degenerate" areas with non-monotonic ordinates.

**Comment by domoritz** [comment]
After the issue is resolved here, make sure that this actually resolves the issues reported in Vega-Lite: https://github.com/vega/vega-lite/issues/4147.

**Comment by jheer** [comment]
Fixed in vega version 4.3.0.
